go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -331,6 +337,12 @@ func (p DoltDatabaseProvider) TableFunction(ctx *sql.Context, name string) (sql.
 	return nil, sql.ErrTableFunctionNotFound.New(name)
 }
 
+// IsRevisionDatabase returns true if the specified dbName represents a database that is tied to a specific
+// branch or commit from a database (e.g. "dolt/branch1").
+func IsRevisionDatabase(dbName string) bool {
```

Comment:
Ah, thanks for pointing out that slashes are valid in db names. It would be a bummer to be more restrictive than MySQL here. That'll require a little more thinking about to handle safely, like you're suggesting. 

I've gone ahead and added a note in the code for now and happy to include this in the database/branch cleanup work this week, too.